src/test/kotlin/com/exactpro/th2/lwdataprovider/TestSseEventSearchRequest.kt

```diff
@@ -0,0 +1,175 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.exactpro.th2.lwdataprovider
+
+import com.exactpro.cradle.TimeRelation
+import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.dataprovider.grpc.EventSearchRequest
+import com.exactpro.th2.lwdataprovider.entities.exceptions.InvalidRequestException
+import com.exactpro.th2.lwdataprovider.entities.requests.SseEventSearchRequest
+import com.exactpro.th2.lwdataprovider.entities.requests.SseMessageSearchRequest
+import com.google.protobuf.Timestamp
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import com.exactpro.th2.dataprovider.grpc.TimeRelation as GrpcTimeRelation
+
+class TestSseEventSearchRequest {
+
+    @Nested
+    inner class TestConstructorParam(){
+        // when startTimestamp and resumeFromId - nulls, should throw exception
+        @Test
+        fun testEmptyParamsMap(){
+            assertThrows<InvalidRequestException>("must throw invalidRequestException"){
+                SseEventSearchRequest(mapOf())
+            }
+        }
+
+        // resumeFromId != null, startTimestamp and endTimestamp - nulls
+        @Test
+        fun testTimestampNullsResumeIdNotNull(){
+            val eventSearchReq = SseEventSearchRequest(mapOf("resumeFromId" to listOf("1")))
+            Assertions.assertNull(eventSearchReq.startTimestamp, "start timestamp must be null")
+            Assertions.assertNull(eventSearchReq.endTimestamp, "end timestamp must be null")
+        }
+
+        // startTimestamp - 1, endTimestamp - 2, searchDirection - AFTER (default)
+        @Test
+        fun testEndAfterStartDirectionDefault(){
+            val eventSearchReq = SseEventSearchRequest(mapOf("startTimestamp" to listOf("1"),
+                "endTimestamp" to listOf("2", "3")))
+            Assertions.assertEquals(TimeRelation.AFTER, eventSearchReq.searchDirection, "search direction must be AFTER")
+            Assertions.assertNotNull(eventSearchReq.startTimestamp, "start timestamp must be not null")
+            Assertions.assertNotNull(eventSearchReq.endTimestamp, "end timestamp must be not null")
+            val messageSupp = {
+                "end timestamp: " + eventSearchReq.endTimestamp + " must be after start timestamp: " + eventSearchReq.startTimestamp
+            }
+            Assertions.assertTrue(eventSearchReq.endTimestamp!! > eventSearchReq.startTimestamp!!, messageSupp)
```

Comment:
You can declare lambda in place
```suggestion
            Assertions.assertTrue(eventSearchReq.endTimestamp!! > eventSearchReq.startTimestamp!!) {
                "end timestamp: " + eventSearchReq.endTimestamp + " must be after start timestamp: " + eventSearchReq.startTimestamp
            }
```

The same for others places